### DISCUSSION TITLE
ABRP connectivity

### DISCUSSION THREAD

**Discussion by andy52h** [tillsteinbach/VWsFriend#413] [Q&A] [answered]
Hi everybody, 

I installed VWsFriend 2 months ago, and I am very happy with all the features. In the Readme, there is a short paragraph on ABRP. The connectivity to VWsFriend allows ABRP to get all the car life data from the car, i.e. from VWsFriend. 

From my understanding, the only thing to connect ABPR to VWsFriend is to generate a 'Generic Token' to get all the live data from VWsFriend. Unfortunately, this doesn't happen, e.g.,  SoC is never updated. 

Does anyone have an idea how to solve the problem?

Regards, 
Andreas

**Comment by thgau** [comment, marked as answer]
ABRP is working for me, but I do not remember exactly how I did the setup. I see some entries in the log of VWsFriend when it calls ABRP: 
INFO:abrp_agent:ABRP send telemetry for vehicle WVGxxxxx for account xxxx@xxxx: Missing telemetry: speed, lat, lon,
(The missing telemetry is because I do not yet have the 3.0 in my ID.4)
As far as I remember: you created the token in ABRP and then go to the settings UI in VWsFriend and enter it there.

**Reply by tillsteinbach** [reply]
Did you configure the Token in the vehicle settings?

**Reply by andy52h** [reply]
Yes, I did. I created the token for OSS and saved the account name and user token in the vehicle settings. Anyway, it works now. Probably it needs some time (several hours) for forwarding the telemetric data, and I was too impatient ...

**Comment by andy52h** [comment]
This is exactly what I did. Anyway, it works now. Probably it needs some time (several hours) for forwarding the telemetric data, and I was too impatient ...
